### ISSUE TITLE
Template parameter metadata description

### ISSUE THREAD

**Issue by BernieWhite** [Azure/azure-resource-manager-schemas#1076]
All throughout Microsoft docs, templates and validation tools such as `arm-ttk` the `metadata.description` property of a template parameter is call out as being valid and recommended.

For example:

```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "storageAccountName": {
            "type": "string",
            "metadata": {
                "description": "The name of the Storage Account."
            }
        }
    },
    "resources": []
}
```

Deployment schemas should reflect that fact that `metadata.description` is a commonly used property by defining it in the schema as _optional_.

This would add the benefit for allowing authoring tools to recognize this is valid property that can be used.

**Comment by BernieWhite** [comment]
@anthony-c-martin Any thoughts?

**Comment by anthony-c-martin** [comment]
I think this a good suggestion - as you mention usage is very widespread and it feels like an unofficial standard. @bmoore-msft thoughts?

**Comment by bmoore-msft** [comment]
One thought we should double-check is where the metadata property is allowed... I *think* it may be allowed anywhere and if we put it only on params that may imply it's not allowed in other areas, which is not ideal.  If the places where metadata is allowed are more finite, we should just add it in those few places...  (though would not add child properties outside of params)

**Comment by BernieWhite** [comment]
@bmoore-msft Yes, unclearly we say metadata can be used anywhere although we demonstrate it being used in:

1. The root of the template, as a peer to `$schema`.
2. As a property of a parameter.
3. At the top level of a resource definition as a peer to `type` and `comments`.
4. Within an output definition as a peer to `type`.

> You can add a metadata object almost anywhere in your template. Resource Manager ignores the object, but your JSON editor may warn you that the property isn't valid. In the object, define the properties you need.

[Metadata reference](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-syntax#metadata)

#1079

**Comment by BernieWhite** [comment]
I raised this a two separate issues but they are interrelated however:

#1076 - `metadata` is a schema defined property of a parameter, `metadata.description` is not but common used and recommended in out docs.
#1079 - `metadata` is not defined in the schema as valid on the template, resources or outputs but is suggested as a valid uses case in our docs.
